### ISSUE TITLE
RFC: Make MergeStrategy-like behavior the normal behavior.

### ISSUE THREAD

**Issue by askreet** [acquia/moonshot#148]
Basically, today we maintain a file on disk that is the source of truth for stack parameters.

I'd like us to move to a place where all parameters in AWS are not updated unless they are explicitly passed. Likewise, we would support non-interactive "answer file" support for creating stacks.

Basically the flows would update to:
## Create
- Determine a list of parameters AWS will need by parsing template.
- Import any Outputs from parent stacks into a hash of parameters.
- Import any values from an answer file, overriding any already set values.
- Add any parameters set with a command-line option, overriding any already set values.
- Prompt the user interactively for any missing parameters.
- Send CreateStack call to AWS.

## Update
- Determine a list of parameters AWS will need by parsing template.
- Look at current stack and set any already set values to UsePreviousValue.
- Import any Outputs from parent stacks that are *missing* (e.g. new values defined in template changes).
- Import any values from an answer file, overriding any already set values.
- Add any parameters set with a command-line option, overriding any already set values.
- Prompt the user interactively for any missing parameters (i.e. defined in this template update).
- Send UpdateStack call to AWS.

In the case of parameters with default values, the user will still be prompted with the default displayed, hitting return will use the default.

## But, what about my CI environment?

We will add support for a --no-interaction option, which will default to true if there is no TTY. When this is set, any missing parameters without defaults will be considered an error, and a non-zero exit code will be returned.

## Mock-ups

```
$ moonshot create
Parameter 'ASGSize' needs value [2]? <RETURN>
Parameter 'InstanceType' needs vallue? t2.micro
Creating stack with parameters:
  ASGSize: 2
  InstanceType: t2.micro
...
```

Later the user wants to update the instance type:
```
$ moonshot update -p InstanceType=t2.small
Updating stack with parameters:
  ASGSize: 2 (Previous Value)
  InstanceType: t2.small
```

If there is a "secret" value with NoEcho, the user would see:
```
$ moonshot update
Updating stack with parameters:
  ASGSize: 2 (Previous Value)
  InstanceType: t2.small (Previous Value)
  SecretAPIKey: (NoEcho set, Previous Value)
```

For interactive use, an answer file can be provided or all parameters can be set on the commandline:
```
$ moonshot create -p ASGSize=2 -p InstanceType=t2.small -p SecretAPIKey=my-secret-key
```

```
cat > answers.yml
---
ASGSize: '2'
InstanceType: t2.small
SecretAPIKey: my-secret-key
^D
$ moonshot create --answers-file answers.yml
```

You can mix and match answers files and `-p`:
```
cat > answers.yml
---
ASGSize: '2'
InstanceType: t2.small
^D
$ moonshot create --answers-file answers.yml -p SecretAPIKey=my-secret-key
```

**Comment by askreet** [comment]
As a future enhancement, the interview process with the human user would do things like multi-choice or user-defined plugins to obtain answers for a given parameter. As a first pass we would define an `AskUser` handler that did the simplest thing.

**Comment by askreet** [comment]
The first part of this was addressed in #151. We still need the interactive interviewing feature.
